### PR TITLE
[Blocked] Add support for streaming non tabular plain text files

### PR DESCRIPTION
**Pull request by OriHoch**
**don't merge, see dependencies section below**

* [x] I've added tests to cover the proposed changes

## overview
In [budgetkey-data-pipelines](https://github.com/OpenBudget/budgetkey-data-pipelines) project we have the need to load some html resourcs and parse them as a stream of data to extract some info from them.

The add_resource and stream_remote_resources are a perfect match for this - they handle the loading of resources from different sources (file / remore) - and streaming the data in a way that can be combined with other pipeline processing.

The only problem is they require tabular data and don't support plain text (or html) content.

## Changes proposed in this pull request:

- modified stream_remote_resources processor to support a "txt" format
- when this format is used - the resource is streamed into tabluar data by splitting into lines and having a single "data" column contain each line
- this was created as a custom Tabulator parser class which extends the csv parser (perhaps it should be included into Tabulator, I'm not sure..)

## dependencies
* This PR depends on the following tabulator-py PR being merged first: https://github.com/frictionlessdata/tabulator-py/pull/154
  * once it's merged, should remove the dependency from Makefile and tox.ini files

### CODE REVIEW

**Comment by OriHoch** [comment]
checking the build failures

**Comment by OriHoch** [comment]
ok, travis should pass now

File: datapackage_pipelines/lib/stream_remote_resources.py [hunk 1]
```diff
@@ -73,14 +74,24 @@ def opener():
...
-                _stream.open()
+                try:
+                    _stream.open()
+                except FormatError as e:
+                    # Tabulator has some validation which is irrelevant for txt format
+                    if _params.get("format") != "txt":
+                        raise
                 _headers = dedupe(_stream.headers)
-                _schema = __resource.get('schema')
+                _schema = _resource.get("schema")
```
Comment (akariv) [inline datapackage_pipelines/lib/stream_remote_resources.py]: This should be __resource, no?

**Comment by akariv** [comment]
@roll this adds a 'text' mode for tabulator - which basically treats a text file as a tabular file with one column ('data'). What do you think about adding this as a built-in parser of tabulator?

**Comment by roll** [comment]
@akariv 
Seems a good idea for `txt` format and `TxtParser`. Anyway it's really dangerouse for `datapackage-pipelines` to rely on `tabulator` internals monkeypatching.

**Comment by OriHoch** [comment]
thanks @roll , I'm checking how the tabulator changes integrate with our use-case

**Comment by akariv** [comment]
Is this still blocked?

**Comment by OriHoch** [comment]
yes, waiting for tabulator merge

**Comment by OriHoch** [comment]
updated, waiting for merge of tabulator pr:
https://github.com/frictionlessdata/tabulator-py/pull/158

File: Makefile [hunk 2]
```diff
@@ -9,6 +9,7 @@ all: list
 
 install:
 	pip install --upgrade -e .[develop]
+	pip install --upgrade https://github.com/frictionlessdata/tabulator-py/archive/custom-loaders-parsers.zip
```
Comment (akariv) [inline Makefile]: We can remove this, right @OriHoch ?

File: tox.ini [hunk 3]
```diff
@@ -18,6 +18,7 @@ passenv=
   TRAVIS_JOB_ID
   TRAVIS_BRANCH
 commands=
+  pip install --upgrade https://github.com/frictionlessdata/tabulator-py/archive/custom-loaders-parsers.zip
```
Comment (akariv) [inline tox.ini]: Also this

File: Makefile (hunk 2 above)
Comment (OriHoch) [inline Makefile]: yes

File: tox.ini (hunk 3 above)
Comment (OriHoch) [inline tox.ini]: yes

**Pull request merged**